libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -504,10 +504,12 @@ void lua_scripts::run(void) {
     // Skip those directores disabled with SCR_DIR_DISABLE param
     uint16_t dir_disable = AP_Scripting::get_singleton()->get_disabled_dir();
     bool loaded = false;
+#if AP_FILESYSTEM_FILE_WRITING_ENABLED
```

Comment:
Its not clear to me that his is correct in every case. Obviously loading a script is a read only operation. Also note that `SCRIPTING_DIRECTORY` is a define that can be overridden.

Reply:
Unfortunately there is no "AP_FILESYSTEM_FILE_OPENDIR_ENABLED", we could define one but it would have the same definition as AP_FILESYSTEM_FILE_WRITING_ENABLED

Reply:
> Unfortunately there is no "AP_FILESYSTEM_FILE_OPENDIR_ENABLED", we could define one but it would have the same definition as 

I'm in favour of `AP_FILESYSTEM_FILE_OPENDIR_ENABLED` rather than reusing  `AP_FILESYSTEM_FILE_WRITING_ENABLED`

However, I'd really rather handle that string emission more gracefully.  So either don't call the load-all-scripts method if the directory doesn't (or can't) exist, or remove the warning from the load-all-scripts method and put it in the places where it is appropriate to warn.

Reply:
That's what this change does - don't call load-all-scripts. In what way would you do this differently.

Reply:
Fixed